torch/ao/quantization/pt2e/quantizer/x86_inductor_quantizer.py

```diff
@@ -369,6 +472,124 @@ def _annotate_conv2d(
+        maxpool_node.meta["quantization_annotation"] = X86InductorQuantizationAnnotation(
+            input_qspec_map=input_qspec_map,
+            _annotated=True,
+        )
+        # Since single maxpool2d decomposed into torch.ops.aten.max_pool2d_with_indices.default - operator.getitem
+        # We annotate it as a fusion pattern.
+        getitem_node.meta["quantization_annotation"] = X86InductorQuantizationAnnotation(
+            _annotated=True,
+            _is_output_of_fusion_pattern=True,
+        )
+
+    def _annotate_quantizable_single_op(
+        self, node: Node, quantization_config: QuantizationConfig
+    ) -> None:
+        # For single op connects to input node of quantized node, insert fake quant between them.
+        if (
+            (node.target in quantizable_ops_pt2e)
+            and (not _is_any_annotated([node]))
+            and (node.op == "call_function")
+        ):
+            def is_all_inputs_connected_to_quantized_op(input_nodes):
+                # Ensure all the inputs connect to fusion pattern or quantized node
+                for input_node in input_nodes:
+                    if not is_quantized_op_pt2e(input_node):
+                        return False
+                return True
+
+            if node.target is torch.ops.aten.max_pool2d_with_indices.default:
+                # Recipe of maxpool2d: check input arg[0] of maxpool2d is quantized or not
+                input_nodes_to_check = [node.all_input_nodes[0]]
+                if not is_all_inputs_connected_to_quantized_op(input_nodes_to_check):
+                    return
+                self._annotate_maxpool2d(node, quantization_config)
+                return
+            else:
+                # TODO <leslie>: Enable recipes for more single quantizable op such as view and relu.
+                pass
+        return
+
+    def _annotate_quantized_node_output(
+        self, node: Node, quantization_config: QuantizationConfig
+    ) -> None:
+        # Recipe refers to https://github.com/intel/intel-extension-for-pytorch/blob/
+        # 90d19323d96afc53fcc22ba5a7bb3fb07fdd6c1c/intel_extension_for_pytorch/quantization/_utils.py#L495
+        r"""
+        1. Force insert observer at output of this node.
+        2. It also fixes this case:
+        For ops which we want to force int8 in and int8 output, there is a case when the op has multi user nodes.
+        If some of the user nodes are not quantizable, we need to lift up obsever from user node's input into output.
+        Otherwise fail to find can't find dq - force_int8_node - quant pattern.
+        # Before
+             observer
+                |
+              view
+              /   \
+        observer non-quantized_op
+           /
+        quantized_op
+        # After
+                observer
+                  |
+                 view
+                  |
+                observer
+                /   \
+        quantized_op non-quantized_op
```

Comment:
May I know how this comment is related to the implementation here?

Reply:
We can remove this comment. It's just about another reason why we need to insert observer at output of some node. Because, it can lift up the position of fake quant. Then, we can find the quantization pattern to match and lowering.